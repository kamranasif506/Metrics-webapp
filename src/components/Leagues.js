import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import LeagueList from './LeagueList';
import DataFilter from './DataFilter';

const Leagues = () => {
  const leagueData = useSelector((store) => store.leagues.leagueData);
  const [filterValue, setFilterValue] = useState('');
  return (
    <section className="container-fluid">
      <div className="row">
        <div className="col-6 m-auto">
          <img
            src={leagueData.logo}
            style={{
              width: '70%', margin: '0 auto', paddingTop: '2%', paddingBottom: '2%',
            }}
            alt={leagueData.name}
            className="img"
          />
        </div>
        <div className="col-6 mx-auto my-auto">
          <h1 className="m-0" style={{ fontWeight: 'bold', color: '#3F1052' }}>{ leagueData.name }</h1>
          <p className="m-0" style={{ color: '#3F1052' }}>
            Country:
            {' '}
            { leagueData.country }
            {' '}
          </p>
          <p className="m-0" style={{ color: '#3F1052' }}>
            Season:
            {' '}
            { leagueData.season }
            {' '}
          </p>
        </div>
      </div>
      <DataFilter onFilter={setFilterValue} />
      <div className="row my-auto" style={{ background: '#3F1052' }}>
        <p className="col-12 my-auto text-white" style={{ paddingLeft: '5%', paddingBottom: '1%' }}>League Rankings</p>
      </div>
      <div className="row">
        {leagueData.standings && leagueData.standings[0] ? (
          leagueData.standings[0]
            .filter(
              (standing) => standing.team.name.toLowerCase().includes(filterValue.toLowerCase()),
            )
            .map((standing) => (
              <LeagueList
                key={standing.team.id}
                id={standing.team.id}
                name={standing.team.name}
                logo={standing.team.logo}
                rank={standing.rank}
              />
            ))
        ) : (
          <p>No standings available</p>
        )}
      </div>
    </section>
  );
};

export default Leagues;
