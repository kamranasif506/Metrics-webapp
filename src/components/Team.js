import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchTeamData } from '../redux/teams/teamSlice';

const Team = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const isLoading = useSelector((store) => store.teams.isLoading);
  const teamData = useSelector((store) => store.teams.teamData);

  useEffect(() => {
    dispatch(fetchTeamData({ id, year: '2022' }));
  }, [dispatch]);
  // console.log(isLoading);
  if (isLoading) {
    return (
      <div>Loading...</div>
    );
  }
  // Use the 'id' to fetch and display league details from an API or data source
  if (Object.keys(teamData).length === 0) {
    return (
      <div style={{
        display: 'flex', alignItems: 'center', justifyContent: 'center', height: '100vh',
      }}
      >
        <h1>No Team data available</h1>
      </div>
    );
  }
  return (
    <section className="container-fluid">
      <div className="row">
        <div className="col-6 m-auto">
          <img
            src={teamData.team.logo}
            style={{
              width: '70%', margin: '0 auto', paddingTop: '2%', paddingBottom: '2%',
            }}
            alt={teamData.team.name}
            className="img"
          />
        </div>
        <div className="col-6 mx-auto my-auto">
          <h1 className="m-0" id="statTitle" style={{ fontWeight: 'bold', color: '#3F1052' }}>{ teamData.team.name }</h1>
          <p className="m-0" style={{ color: '#3F1052' }}>
            Rank:
            {' '}
            { teamData.rank }
            {' '}
          </p>
          <p className="m-0" style={{ color: '#3F1052' }}>
            Points:
            {' '}
            { teamData.points }
            {' '}
          </p>

        </div>
      </div>
      <section id="all-matches" className="stats">
        <div className="row my-auto" style={{ background: '#3F1052' }}>
          <p className="col-12 my-auto text-white" style={{ paddingLeft: '3%', paddingBottom: '1%' }}>All Matches</p>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Played</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.all.played }</p>
          </div>
        </div>

        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Won</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.all.win }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Lose</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.all.lose }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Draw</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.all.draw }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">GF</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.all.goals.for }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">GA</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.all.goals.against }</p>
          </div>
        </div>
      </section>
      <section id="home-matches" className="stats">
        <div className="row my-auto" style={{ background: '#3F1052' }}>
          <p className="col-12 my-auto text-white" style={{ paddingLeft: '3%', paddingBottom: '1%' }}>Home Matches</p>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Played</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.home.played }</p>
          </div>
        </div>

        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Won</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.home.win }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Lose</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.home.lose }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Draw</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.home.draw }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">GF</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.home.goals.for }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">GA</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.home.goals.against }</p>
          </div>
        </div>
      </section>
      <section id="away-matches" className="stats">
        <div className="row my-auto" style={{ background: '#3F1052' }}>
          <p className="col-12 my-auto text-white" style={{ paddingLeft: '3%', paddingBottom: '1%' }}>Away Matches</p>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Played</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.away.played }</p>
          </div>
        </div>

        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Won</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.away.win }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Lose</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.away.lose }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">Draw</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.away.draw }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">GF</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.away.goals.for }</p>
          </div>
        </div>
        <div className="row" style={{ borderBottom: '1px solid #53116E' }}>
          <div className="col-6 d-flex">
            <p className="my-auto">GA</p>
          </div>
          <div className="col-6 d-flex justify-content-end">
            <p className="my-auto">{ teamData.away.goals.against }</p>
          </div>
        </div>
      </section>
    </section>
  );
};

export default Team;
