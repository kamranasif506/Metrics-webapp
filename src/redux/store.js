import { configureStore } from '@reduxjs/toolkit';
import leagueReducer from './league/leagueSlice';

const store = configureStore({
  reducer: {
    leagues: leagueReducer,
  },
});

export default store;
