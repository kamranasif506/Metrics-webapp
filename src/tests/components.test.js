import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Leagues from '../components/Leagues';
import Team from '../components/Team';

describe('Component testing', () => {
  test('League component should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Leagues />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Team component should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Team />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
