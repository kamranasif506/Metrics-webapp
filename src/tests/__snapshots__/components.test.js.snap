// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing League component should render correctly 1`] = `
<DocumentFragment>
  <section
    class="container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="col-6 m-auto"
      >
        <img
          class="img"
          style="width: 70%; margin: 0px auto; padding-top: 2%; padding-bottom: 2%;"
        />
      </div>
      <div
        class="col-6 mx-auto my-auto"
      >
        <h1
          class="m-0"
          style="font-weight: bold; color: rgb(63, 16, 82);"
        />
        <p
          class="m-0"
          style="color: rgb(63, 16, 82);"
        >
          Country:  
        </p>
        <p
          class="m-0"
          style="color: rgb(63, 16, 82);"
        >
          Season:  
        </p>
      </div>
    </div>
    <div>
      <div
        class="row my-2"
      >
        <div
          class="col-12"
        >
          <input
            class="form-control w-100 white-placeholder-input"
            placeholder="Filter team..."
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="row my-auto"
      style="background: rgb(63, 16, 82);"
    >
      <p
        class="col-12 my-auto text-white"
        style="padding-left: 5%; padding-bottom: 1%;"
      >
        League Rankings
      </p>
    </div>
    <div
      class="row"
    >
      <p>
        No standings available
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Component testing Team component should render correctly 1`] = `
<DocumentFragment>
  <div
    style=""
  >
    Loading...
  </div>
</DocumentFragment>
`;
